manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/BridgeWorker.java

```diff
@@ -0,0 +1,93 @@
+package com.redhat.service.bridge.manager.workers.resources;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.manager.RhoasService;
+import com.redhat.service.bridge.manager.dao.BridgeDAO;
+import com.redhat.service.bridge.manager.models.Bridge;
+import com.redhat.service.bridge.manager.models.Work;
+import com.redhat.service.bridge.manager.providers.InternalKafkaConfigurationProvider;
+import com.redhat.service.bridge.rhoas.RhoasTopicAccessType;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+import io.quarkus.vertx.ConsumeEvent;
+
+@ApplicationScoped
+public class BridgeWorker extends AbstractWorker<Bridge> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BridgeWorker.class);
+
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    RhoasService rhoasService;
+
+    @Inject
+    InternalKafkaConfigurationProvider internalKafkaConfigurationProvider;
+
+    @Override
+    protected PanacheRepositoryBase<Bridge, String> getDao() {
+        return bridgeDAO;
+    }
+
+    // This must be equal to the Bridge.class.getName()
+    @ConsumeEvent(value = "com.redhat.service.bridge.manager.models.Bridge", blocking = true)
+    public Bridge handleWork(Work work) {
+        return super.handleWork(work);
+    }
+
+    @Override
+    public Bridge createDependencies(Work work, Bridge bridge) {
+        LOGGER.info("Creating dependencies for '{}' [{}]",
+                bridge.getName(),
+                bridge.getId());
+
+        // This call is blocking *BUT* this Worker is run asynchronously by VertX. The Shard will therefore potentially
+        // detect that there is a Bridge in ACCEPTED state and start provisioning it in Kubernetes before this call completes.
+        // Options:
+        // (1) Make "KafkaTopic" a "first class dependency" (i.e. ManagedResource), DB table etc etc so we can block
+        //     the Shard from creating the Bridge until this has completed successfully. This would also be a change
+        //     to ConnectorWorker to depend on the "KafkaTopicManagedResource". TBH this seems a good idea. We'd
+        //     have a DB table for ALL KafkaTopics we've created and hence be able to clean-up better...
+        //     Which is then a leading question for Luca's idea of a generic "Resource" table vs discrete ones
+        //     for Bridge, Processor, Connector, KafkaTopics....
+        // (2) Close JIRA MGDOBR-363, don't have Bridges provisioned with Workers. Use of RHOAS then lacks
+        //     support of a re-try or timeout mechanism, provisioning of resources (i.e. KafkaTopic) for Bridges
+        //     looses any resilience; blah blah blah.
```

Comment:
Hey @lucamolteni @robpblake @danielezonca Don't be afraid of this PR.. it's a draft and this comment I want to discuss with you..... My opinion is to make "KafkaTopic" a first class `ManagedResource` so we can have the `Worker` mechanism treat it as a fallible dependency and hence be subject to re-tries etc.

This JIRA is for Sprint 216 but before I go running off thought to ask your opinions.. happy to set up a meeting for a chat if you like :smile: :coffee: :computer:

Reply:
I'm all for it, eventually I'd like us to provision all kinds of resources in the same way

Reply:
@manstis @lucamolteni I'm not following the discussion here. Why is using the worker pattern to provision the topic/service account for a bridge with our current implementation a problem?

In particular this comment:

```
 The Shard will therefore potentially
        // detect that there is a Bridge in ACCEPTED state and start provisioning it in Kubernetes before this call completes.
```
Our query to expose Bridges to provision to the shard will be `from bridge b where b.status="AVAILABLE" and b.dependencyStatus="READY"` ? What am I missing?

Reply:
I'm not following either @robpblake I think this is a good idea. Where did you find that comment?

Reply:
I can't even begin to explain how my mind was working when I wrote that comment :rofl: 

I'd clearly forgotten about `dependencyStatus`.

Anyway, PR updated and ready for your reviews.